Darkly/LDUserBuilder.m

```diff
@@ -228,8 +235,22 @@ -(User *)build
         [user setAnonymous:currentAnonymous];
     }
     DEBUG_LOG(@"LDUserBuilder building User with anonymous: %d", anonymous);
-    [[DataManager sharedManager] saveContext];
+    [self saveUser: user];
     return user;
 }
 
+-(void) saveUser: (User *) user {
+    [[DataManager sharedManager] purgeOldUsers];
```

Comment:
I think we only need to call purgeOldUsers if we are saving a new User, not also in the case where we are updating a user.